src/com/foamdev/demos/addHTMLElement/Controller.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+// Add HTML element to the view
+
+foam.CLASS({
+  package: 'com.foamdev.demos.addHTMLElement',
+  name: 'Controller',
+  extends: 'foam.u2.Controller',// it will automatically exports: ['as data']
+  //extends: 'foam.u2.Element',
+
+  //exports: [ 'as data' ], //FOAM has DI (Dependency Injection) built-in at the object level.
+
+  css: `
+    body, input[text] { color: #888; font-family: Cambria, Georgia; }
+  `,
+
+  methods: [
+    function initE() {
+      this.
+        start(this.ADD_ELEMENT_TO_DSL, {label: 'add element to the HTML using the DSL'}).end();
+    }
+  ],
+  actions: [
+    {
+      name:  'addElementToDSL',
+      label: 'add a DSL',

```

Comment:
I think most users will find the term DSL confusing or foreign. We should use the term DOM or virtual-DOM.